software/winrm/src/main/java/org/apache/brooklyn/location/winrm/WinRmMachineLocation.java

```diff
@@ -197,7 +198,15 @@ public String getHostname() {
     @Nullable
     protected String getHostAndPort() {
         String host = getHostname();
-        return (host == null) ? null : host + ":" + config().get(WINRM_PORT);
+        return (host == null) ? null : host + ":" + (config().get(USE_SECURE_WINRM) ? 5986 : 5985);
+    }
+
+    public int getPort() {
+        if (getConfig(WINRM_PORT) != null) {
+            return getConfig(WINRM_PORT);
+        } else {
+            return getConfig(USE_SECURE_WINRM) ? 5986 : 5985;
```

Comment:
From the location config from the blueprint.
If you set a location config `winrm.useHttps` to the JcloudsLocation from the blueprint, the config is inherited for the MachineLocation.